### PR TITLE
Fix: absolute import handling

### PR DESCRIPTION
**Pull request by KPilnacek**
Fixes: https://github.com/cython/cython/issues/2638

Though still a test is missing.

### CODE REVIEW

**Comment by scoder** [comment]
Thanks. Looks like Jenkins is unhappy about a couple of tests where the regex fails to match.

**Comment by KPilnacek** [comment]
Sorry, I did not take a look before.

I hopefully fixed it. I'll let Jenkins do its job now and check it in the afternoon.

File: Cython/Build/Dependencies.py [hunk 1]
```diff
@@ -488,6 +490,13 @@ def parse_dependencies(source_filename):
         cimport_from, cimport_list, extern, include = m.groups()
         if cimport_from:
             cimports.append(cimport_from)
+            m_after_from = dependency_after_from_regex.search(source[m.end():])
```
Comment (scoder) [inline Cython/Build/Dependencies.py]: `regex.search()` accepts a start position as second argument. That way, we can avoid to slice and copy the source string here.

**Comment by scoder** [comment]
Looks nice. The travis failure is due to a broken test in master that I fixed now.

File: Cython/Build/Dependencies.py (hunk 1 above)
Comment (KPilnacek) [reply Cython/Build/Dependencies.py]: Thanks for the suggestion - didn't know about that. Changed it in a new fixup.

**Comment by KPilnacek** [comment]
So I rebased it on actual master, hopefully the tests will be OK now.

**Comment by scoder** [comment]
Thanks!

**Comment by KPilnacek** [comment]
I was a bit desperate about the failing test so thanks for fixing that. And thanks for your time :-)

**Comment by jdemeyer** [comment]
What's the rationale for removing this:
```diff
-            elif not self.quiet:
-                print("%s: cannot find cimported module '%s'" % (filename, module))
```
That still seems like a useful warning message...

**Comment by KPilnacek** [comment]
@jdemeyer with this PR, the importing procedure tries all combinations of packages and modules from `from package cimport module`. However, this approach creates false positives if the part after `cimport` is not a module but an object from the package. This print statements would then emit all failures of attempted imports `package.object`.

On the other hand, the PRed procedure of generating all possible imports should be robust enough to capture all imports. And thus, this shouldn't be necessary as there should not be any false negatives.

**Comment by jdemeyer** [comment]
> And thus, this shouldn't be necessary as there should not be any false negatives.

... except if there are bugs. With that reasoning, you might as well remove all `assert` statements everywhere.

**Comment by scoder** [comment]
@jdemeyer, note that this only changes the dependency tracking to decide whether a module needs to be recompiled. Finding too many candidates is ok here. (Ok, stat-calls are not for free, but they are also not very expensive.) _Missing_ a dependency would be a problem, because it might prevent a necessary rebuild.

In any case, Cython will still do a proper lookup of all compile-time dependencies later during the compilation, and if any of them is not found, you will get a compiler error.

**Comment by jdemeyer** [comment]
> _Missing_ a dependency would be a problem, because it might prevent a necessary rebuild.

Right, and this PR removes a warning message if that happens.

**Pull request merged**